### PR TITLE
Update

### PR DESCRIPTION
**Pull request by madeindjs**
Dear @jsahlen ,

I just completely upgraded your gem (after 6 years of inactivity). I made theses actions:

- update Rake & Rspec (from 2.* to 3.*)
- update all Rspec tests with new syntax `expect(..).to eq ..`
- pass Rubocop parser on somes files to respect syntax conventions
- correct a failing text on  347df2b 

I hope you'll accept this PR cauz' I want to implement some new stuff on your library.

### CODE REVIEW

**Comment by madeindjs** [comment]
@jsahlen , is this project alive?

**Comment by jsahlen** [comment]
@madeindjs "Alive" is a strong word, but, yes! I'm grateful for your contribution, I just haven't had the time to have a thorough look yet—sorry.

File: lib/todotxt.rb [hunk 1]
```diff
@@ -2,14 +2,16 @@
 libdir = File.dirname(__FILE__)
 $LOAD_PATH.unshift(libdir) unless $LOAD_PATH.include?(libdir)
 
+
+# Todotxt is a Ruby librairy / CLI interface to work with a [todo.txt](http://www.todotxt.com) format.
```
Comment (jsahlen) [inline lib/todotxt.rb]:
```diff
- # Todotxt is a Ruby librairy / CLI interface to work with a [todo.txt](http://www.todotxt.com) format.
+ # Todotxt is a Ruby librairy / CLI interface to work with [todo.txt](http://www.todotxt.com) format files.
```

File: lib/todotxt/cli.rb [hunk 2]
```diff
@@ -1,42 +1,41 @@
...
       # Open config file and render config.
       @config = Config.new options
       @list   = nil
-      unless ["help", "generate_config", "generate_txt"].include? ARGV[0]
+      unless %w[help generate_config generate_txt].include? ARGV[0]
         ask_and_create @config unless @config.file_exists?
-        if @config.deprecated? and options[:file]
-          error_and_exit "You are using an old config, which has no support for mulitple files. Please update your configuration."
+        if @config.deprecated? && options[:file]
+          error_and_exit 'You are using an old config, which has no support for mulitple files. Please update your configuration.'
```
Comment (jsahlen) [inline lib/todotxt/cli.rb]:
```diff
- You are using an old config, which has no support for mulitple files. Please update your configuration.
+ You are using an old config, which has no support for multiple files. Please update your configuration.
```

File: lib/todotxt/todo.rb [hunk 3]
```diff
@@ -1,58 +1,59 @@
-require "todotxt/regex"
+require 'todotxt/regex'
 
 module Todotxt
+  # Represent a task formated according
```
Comment (jsahlen) [inline lib/todotxt/todo.rb]:
```diff
- Represent a task formated according
+ Represent a task formatted according to
```

File: lib/todotxt/todo.rb [hunk 4]
```diff
@@ -1,58 +1,59 @@
...
+  # [todo.txt format rules](https://github.com/todotxt/todo.txt#todotxt-format-rules)
+  #
+  # @attr [String] text reprensent all text definition of this task
```
Comment (jsahlen) [inline lib/todotxt/todo.rb]:
```diff
- reprensent all text definition of this task
+ the complete text definition of this task
```

File: lib/todotxt/todo.rb [hunk 5]
```diff
@@ -1,58 +1,59 @@
...
+  # @attr [Integer] line the line number of this task
+  # @attr [Char] priority the letter who define pritority
```
Comment (jsahlen) [inline lib/todotxt/todo.rb]:
```diff
- the letter who define pritority
+ the character which defines the task's priority
```

File: lib/todotxt/todo.rb [hunk 6]
```diff
@@ -1,58 +1,59 @@
...
 
+    # Mark this task as done
     def do
       unless done
         @text = "x #{text}".strip
         @done = true
       end
     end
 
+    # Mark this task as undone
```
Comment (jsahlen) [inline lib/todotxt/todo.rb]:
```diff
- Mark this task as undone
+ Mark this task as not done
```

File: lib/todotxt/todo.rb [hunk 7]
```diff
@@ -63,33 +64,46 @@ def prioritize new_priority=nil, opts={}
       @priority = new_priority
     end
 
-    def append appended_text=""
-      @text << " " << appended_text
+    # Add some content to the task
```
Comment (jsahlen) [inline lib/todotxt/todo.rb]:
```diff
- Add some content to the task
+ Add some text to the end of this task definition
```

File: lib/todotxt/todo.rb [hunk 8]
```diff
@@ -63,33 +64,46 @@ def prioritize new_priority=nil, opts={}
...
+    def append(appended_text = '')
+      @text << ' ' << appended_text
     end
 
-    def prepend prepended_text=""
+    # Add some content to the task
```
Comment (jsahlen) [inline lib/todotxt/todo.rb]:
```diff
- Add some content to the task
+ Add some text to the beginning of this task definition
```

File: lib/todotxt/todo.rb [hunk 9]
```diff
@@ -63,33 +64,46 @@ def prioritize new_priority=nil, opts={}
...
+    # @return [String]
     def to_s
       text.clone
     end
 
-    def <=> b
-      if priority.nil? && b.priority.nil?
-        return line <=> b.line
-      end
+    # Compare another `Todo` according to their `line` and `priority` attribute.
```
Comment (jsahlen) [inline lib/todotxt/todo.rb]:
```diff
- Compare another `Todo` according to their `line` and `priority` attribute.
+ Compare with another `Todo` based on `line` and `priority` attributes
```

File: lib/todotxt/todo.rb [hunk 10]
```diff
@@ -63,33 +64,46 @@ def prioritize new_priority=nil, opts={}
...
 
+    private
+
+    def create_from_text(text)
+      @text = text
+      @priority = text.scan(PRIORITY_REGEX).flatten.first || nil
+      @projects = text.scan(PROJECT_REGEX).flatten.uniq   || []
+      @contexts = text.scan(CONTEXT_REGEX).flatten.uniq   || []
+      @done = !text.scan(DONE_REGEX).empty?
+    end
```
Comment (jsahlen) [inline lib/todotxt/todo.rb]: 👍

File: lib/todotxt/todofile.rb [hunk 11]
```diff
@@ -1,34 +1,31 @@
...
     # Generate a file from template
     def generate!
-      FileUtils.copy File.join(File.dirname(File.expand_path(__FILE__)), "..", "..", "conf", "todo.txt"), @path
+      FileUtils.copy File.join(File.dirname(File.expand_path(__FILE__)), '..', '..', 'conf', 'todo.txt'), @path
     end
 
-    def path
-      @path
-    end
+    attr_reader :path
```
Comment (jsahlen) [inline lib/todotxt/todofile.rb]: 👍

File: lib/todotxt/todolist.rb [hunk 12]
```diff
@@ -1,17 +1,18 @@
-require "todotxt/todo"
+require 'todotxt/todo'
 
 module Todotxt
-  #@TODO merge with TodoFile, both overlap too much
+  # Represent a collection of `Todo`
```
Comment (jsahlen) [inline lib/todotxt/todolist.rb]:
```diff
- Represent a collection of `Todo`
+ Represent a collection of `Todo` items
```

File: lib/todotxt/todolist.rb [hunk 13]
```diff
@@ -21,44 +22,51 @@ def initialize file, line = nil
...
       @todos.reject! { |t| t.line.to_s == line.to_s }
     end
 
-    def move line, other_list
+    def move(line, other_list)
       other_list.add find_by_line(line).to_s
       remove line
     end
 
+    # Get all existing projects into todos
```
Comment (jsahlen) [inline lib/todotxt/todolist.rb]:
```diff
- Get all existing projects into todos
+ Get all projects from todo definitions
```

File: lib/todotxt/todolist.rb [hunk 14]
```diff
@@ -21,44 +22,51 @@ def initialize file, line = nil
...
+    # @return[Array<String>]
     def projects
-      map { |t| t.projects }.flatten.uniq.sort
+      map(&:projects).flatten.uniq.sort
     end
 
+    # Get all existing contects into todos
```
Comment (jsahlen) [inline lib/todotxt/todolist.rb]:
```diff
- Get all existing contects into todos
+ Get all contexts from todo definitions
```

**Review by jsahlen — CHANGES_REQUESTED** [review]
Thanks so much for this! I had some spelling- and language-related comments (and I'm aware some of the issues were there before your patches), but other than that, looks great!

**Comment by madeindjs** [comment]
@jsahlen , I made your proposed corrections in a5a9791

**Review by jsahlen — APPROVED** [review]

**Comment by jsahlen** [comment]
@madeindjs Thanks! I've published the new version to RubyGems.org

**Pull request merged**